src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -62,255 +83,356 @@ interface DriveExplorerProps {
   connectDropTarget: ConnectDropTarget;
 }
 
-interface DriveExplorerState {
-  fileInputRef: React.RefObject<HTMLInputElement>;
-  fileInputKey: number; //! Changing this forces the invisible file input to render
-  email: string;
-  token: string;
-  isAdmin: boolean;
-  isMember: boolean;
-}
-
-class DriveExplorer extends Component<DriveExplorerProps, DriveExplorerState> {
-  constructor(props: DriveExplorerProps) {
-    super(props);
-
-    this.state = {
-      fileInputRef: createRef(),
-      fileInputKey: Date.now(),
-      email: '',
-      token: '',
-      isAdmin: true,
-      isMember: false,
-    };
-  }
+const DriveExplorer = (props: DriveExplorerProps): JSX.Element => {
+  const dispatch = useAppDispatch();
+
+  const [fileInputRef] = useState<RefObject<HTMLInputElement>>(createRef());
+  const [fileInputKey, setFileInputKey] = useState<number>(Date.now());
+  const [folderInputRef] = useState<RefObject<HTMLInputElement>>(createRef());
+  const [folderInputKey, setFolderInputKey] = useState<number>(Date.now());
+
+  const {
+    selectedItems,
+    isLoading,
+    viewMode,
+    title,
+    titleClassName,
+    items,
+    isDeleteItemsDialogOpen,
+    isCreateFolderDialogOpen,
+    onItemsDeleted,
+    onFolderCreated,
+    isOver,
+    connectDropTarget,
+    storageFilters,
+    currentFolderId,
+    onFileUploaded
+  } = props;
 
-  get hasAnyItemSelected(): boolean {
-    return this.props.selectedItems.length > 0;
-  }
+  useEffect(() => {
+    deviceService.redirectForMobile();
+  }, []);
 
-  get hasItems(): boolean {
-    return this.props.items.length > 0;
-  }
+  const hasAnyItemSelected = (): boolean => {
```

Comment:
Since this.props.items changes on every render you don't have to declare them as functions but as values like:

`const hasItems = this.props.items.length >0;`

Same for `hasAnyItemSelected`, `hasItems` and `hasFilters`